yt_dlp/extractor/piapro.py

```diff
@@ -0,0 +1,97 @@
+# coding: utf-8
+
+from __future__ import unicode_literals
+from .common import InfoExtractor
+from ..compat import compat_urlparse
+from ..utils import int_or_none, parse_duration, parse_filesize, unified_timestamp, urlencode_postdata, ExtractorError
+
+
+class PiaproIE(InfoExtractor):
+    _NETRC_MACHINE = 'piapro'
+    _VALID_URL = r'https?://piapro\.jp/t/(?P<id>\w+)/?'
+    _TESTS = [{
+        'url': 'https://piapro.jp/t/NXYR',
+        'md5': 'a9d52f27d13bafab7ee34116a7dcfa77',
+        'info_dict': {
+            'id': 'NXYR',
+            'ext': 'mp3',
+            'uploader': 'wowaka',
+            'uploader_id': 'wowaka',
+            'title': '裏表ラバーズ',
+            'thumbnail': r're:^https?://.*\.jpg$',
+        }
+    }]
+
+    def _real_initialize(self):
+        self._login_status = self._login()
+
+    def _login(self):
+        username, password = self._get_login_info()
+        if not username:
+            return False
+        login_ok = True
+        login_form_strs = {
+            '_username': username,
+            '_password': password,
+            '_remember_me': 'on',
+            'login': 'ログイン'
+        }
+        self._request_webpage('https://piapro.jp/login/', None)
+        urlh = self._request_webpage(
+            'https://piapro.jp/login/exe', None,
+            note='Logging in', errnote='Unable to log in',
+            data=urlencode_postdata(login_form_strs))
+        if urlh is False:
+            login_ok = False
+        else:
+            parts = compat_urlparse.urlparse(urlh.geturl())
+            if parts.path != '/':
+                login_ok = False
+        if not login_ok:
+            self._downloader.report_warning(
```

Comment:
`self.report_warning`.

Actually this can be an error

Reply:
oops
I don't think it can be an error, as a convention it seems to be warnings are used

Reply:
If the extractor can function without login, use warnings. Otherwise, it can be error